props/saber_BC_buttons.h

```diff
@@ -477,738 +1478,1227 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
-      if (a > M_PI / 6) VolumeUp();
-      if (a < -M_PI / 6) VolumeDown();
-    } else if (scroll_presets_) {
-
-      if (a > M_PI / 6) {
-        beeper.Beep(0.05, 4000);
-        current_menu_angle_ = fusor.angle2();
-        next_preset();
-        }
-
-      if (a < -M_PI / 6) {
-        beeper.Beep(0.05, 3000);
-        current_menu_angle_ = fusor.angle2();
-        previous_preset();
-      }
-    }
+  void BeepExitFeature() {
+    beeper.Beep(0.05, 3000);
+    beeper.Silence(0.05);
+    beeper.Beep(0.05, 3000);
+    beeper.Silence(0.05);
+    beeper.Beep(0.10, 2000);
   }
 
-  bool is_pointing_up() {
+  bool isPointingUp() {
     if (fusor.angle1() > M_PI / 3) return true;
-#ifdef DUAL_BLADES
+#ifdef BC_DUAL_BLADES
     if (fusor.angle1() < -M_PI / 3) return true;
 #endif
     return false;
   }
 
   void TurnOnHelper() {
-    if (is_pointing_up()) {
+    if (isPointingUp() || muted_) {
       FastOn();
     } else {
       On();
     }
   }
 
   void TurnOffHelper() {
-      if (is_pointing_up()) {
+    if (SaberBase::Lockup() || battle_mode_) return;
+    if (isPointingUp()) {
       Off(OFF_FAST);
     } else {
       Off();
     }
     saber_off_time_ = millis();
     battle_mode_ = false;
+    spam_blast_ = false;
+    muted_ = false;
   }
 
-  void DoMute() {
-    if (SetMute(true)) {
-      unmute_on_deactivation_ = true;
-      TurnOnHelper();
+  void MuteAll() {
+    if (hybrid_font.PlayPolyphonic(&SFX_mute)) {
+      mute_all_timer_.trigger(SaberBase::sound_length * 1000);
+    } else {
+      if (SetMute(true)) {
+        unmute_on_deactivation_ = true;
+        muted_ = true;
+        TurnOnHelper();
+      }
     }
   }
 
-  RefPtr<BufferedWavPlayer> wav_player;
-
-  bool Event2(enum BUTTON button, EVENT event, uint32_t modifiers) override {
-    switch (EVENTID(button, event, modifiers)) {
-      case EVENTID(BUTTON_AUX, EVENT_PRESSED, MODE_ON):
-      case EVENTID(BUTTON_AUX2, EVENT_PRESSED, MODE_ON):
-      return true;
-
-// Gesture Ignition Controls
-#ifdef BC_SWING_ON
-    case EVENTID(BUTTON_NONE, EVENT_SWING, MODE_OFF):
-      if (mode_volume_) return false;  
-#ifdef NO_BLADE_NO_GEST_ONOFF
-      if (!blade_detected_) return false;
-#endif
-        FastOn();
-#ifdef BC_GESTURE_AUTO_BATTLE_MODE
-        PVLOG_NORMAL << "Entering Battle Mode\n";
-        battle_mode_ = true;
-#endif
-      return true;
-#endif  // BC_SWING_ON
-
-#ifdef BC_TWIST_ON
-    case EVENTID(BUTTON_NONE, EVENT_TWIST, MODE_OFF):
-      if (mode_volume_ || scroll_presets_) return false;
-#ifdef NO_BLADE_NO_GEST_ONOFF
-      if (!blade_detected_) return false;
-#endif
-      // Delay twist events to prevent false trigger from over twisting
-      if (millis() - last_twist_ > 2000 &&
-        millis() - saber_off_time_ > 1000) {
-        FastOn();
-#ifdef BC_GESTURE_AUTO_BATTLE_MODE
-        PVLOG_NORMAL << "Entering Battle Mode\n";
-        battle_mode_ = true;
-#endif
-        last_twist_ = millis();
+  void DoChangePreset() {
```

Comment:
Well it's doing previous, next, or first preset, so I had thought DoChangePreset made sense.
Would you prefer DoChoosePreset? 
I'm open to suggestions of course.

Reply:
How about this?

Reply:
> How about this?

What changed?

> I could also just add "// Previous, next, or first preset, depending on blade angle"

That seems like a useful comment, please add.

Reply:
Sorry, that was a "bump" to bring attention to an unresolved comment.
I meant to ask if any of those alternate names were better, or if the comment is enough.
I've added the comment for now, please let me know otherwise.